modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-channel-impl/src/main/java/com/liferay/headless/commerce/admin/channel/internal/resource/v1_0/CategoryDisplayPageResourceImpl.java

```diff
@@ -165,14 +191,37 @@ public CategoryDisplayPage postChannelIdCategoryDisplayPage(
 			Long id, CategoryDisplayPage categoryDisplayPage)
 		throws Exception {
 
+		long categoryId = GetterUtil.getLong(
+			categoryDisplayPage.getCategoryId());
+
+		if (categoryId == 0) {
+			Group group = _groupService.fetchGroupByExternalReferenceCode(
+				categoryDisplayPage.getGroupExternalReferenceCode(),
+				contextCompany.getCompanyId());
+
+			if (group == null) {
+				throw new NoSuchGroupException();
```

Comment:
I'm not sure we should be throwing exception in cases like this. can you please check with other APIs?